### PR TITLE
HURUmap visual as posttype

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Desktop Screenshots

Screen Shot 2019-12-06 at 4 29 52 PM

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: wp-content/plugins/hurumap/api.php [hunk 1]
```diff
@@ -90,204 +44,89 @@ function register_routes()
...
-
-    if ($chart_id) {
-        $hurumap = $wpdb->get_results($wpdb->prepare("SELECT * FROM {$wpdb->base_prefix}hurumap_charts where published = 1 and id = %s", [$chart_id]));
-        $section = array();
-        if ($hurumap[0]) {
-            $section = $wpdb->get_results($wpdb->prepare("SELECT * FROM {$wpdb->base_prefix}chart_sections where published = 1 and id = %s", [$hurumap[0]->section]));
+    $id = $request->get_param('chart_id');
+    if ($id) {
+        $post = get_post($id);
+        $chart = json_decode($post->post_content, true);
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/api.php:65]: Lets confirm `post_type` is `hurumap-visual` before proceeding.

File: wp-content/plugins/hurumap/api.php [hunk 2]
```diff
@@ -90,204 +44,89 @@ function register_routes()
...
-        }
-    } else {
-        update_or_create_section($json);
+    $posts = get_posts(array(
+        'posts_per_page'			=> -1,
+        'post_type'					=> 'hurumap-section',
+        'post_status'				=> array('publish')
+    ));
+    $sections = array();
+    foreach( $posts as $post ) {
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/api.php]: If no sections have been defined yet, `$posts` will be empty and empty array will be returned _even if we have charts..._

File: wp-content/plugins/hurumap/api.php [hunk 3]
```diff
@@ -90,204 +44,89 @@ function register_routes()
...
-
     WP_Filesystem();
 
-    $chart_id = $request->get_param('chart_id');
-    $flourish = $wpdb->get_results($wpdb->prepare("SELECT * FROM {$wpdb->base_prefix}flourish_charts where id=%s  LIMIT 1", $chart_id));
-    $file_content = $flourish[0]->file;
+    $id = $request->get_param('chart_id');
+    $post = get_post($id);
+    if ($post) {
+        $content = json_encode($post->post_content);
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/api.php]: Again need to at least check if it's `hurumap-visual` post (and it's actually of type `flourish`)

File: wp-content/plugins/hurumap/api.php [hunk 4]
```diff
@@ -90,204 +44,89 @@ function register_routes()
...
+        $file_id = $content->fileId;
+    }
 
     //assign directory and file
-    $destination_dir = "flourish/zip" . $chart_id . "/";
-    $chart_zip_path = get_attached_file((int) $flourish[0]->media_id);
+    $destination_dir = "flourish/zip" . $id . "/";
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/api.php]: Any reason not to create this flourish folder (or just the `zip.$id."/"` folder) in the system temporary folder `sys_get_temp_dir()` ?

File: wp-content/plugins/hurumap/helpers.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	
+	// bail early if not defined
+	if( !function_exists('get_current_screen') ) {
+		return false;
+	}
+	
+	// vars
+	$current_screen = get_current_screen();
+	
+	echo "<script>console.log('{$current_screen->id}');</script>";
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/helpers.php]: ?

File: wp-content/plugins/hurumap/data.php [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+                    'delete_post'        => $cap,
+                    'edit_posts'        => $cap,
+                    'delete_posts'        => $cap,
+                ),
+                'supports'             => array('title'),
+                'rewrite'            => false,
+                'query_var'            => false,
+            ));
+
+            //hook into the init action and call create_book_taxonomies when it fires
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/data.php]: ?

File: wp-content/plugins/hurumap/helpers.php [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	
+	// array
+	} elseif( is_array($id) ) {
+		return in_array($current_screen->id, $id);
+	
+	// string
+	} else {
+		return ($id === $current_screen->id);
+	}
+}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/helpers.php]: `\n`

File: wp-content/plugins/hurumap/index.php [hunk 8]
```diff
@@ -12,104 +12,238 @@
...
+    return $hurumap;
 }
 
-add_action('admin_enqueue_scripts', 'register_admin_scripts');
-add_action('admin_enqueue_scripts', 'load_admin_scripts');
-add_action('admin_menu', 'setup_admin_menu');
\ No newline at end of file
+hurumap();
+
+endif;
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/index.php]: `\n`

File: wp-content/plugins/hurumap/package.json [hunk 9]
```diff
@@ -30,11 +30,13 @@
...
+    "start": "concurrently \"yarn start:definitions\"  \"yarn start:blocks\"",
     "start:blocks": "wp-scripts start ./src/FlourishBlock/index.js ./src/HURUmapBlock/index.js ./src/FeaturedData/index.js ./src/CardBlock/index.js --output-path=build/blocks",
-    "start:data-page": "wp-scripts start ./src/index.js --output-path=build/data-page",
-    "build": "yarn build:data-page && yarn build:blocks && yarn build:micro-frontend",
-    "build:data-page": "wp-scripts build ./src/index.js --output-path=build/data-page",
+    "start:definitions": "wp-scripts start ./src/ChartDefinition/index.js --output-path=build/definitions",
+    "start:sections": "wp-scripts start ./src/ChartsSection/index.js --output-path=build/sections",
+    "build": "yarn build:definitions && yarn build:blocks && yarn build:micro-frontend",
+    "build:definitions": "wp-scripts build ./src/ChartDefinition/index.js --output-path=build/definition",
+    "build:sections": "wp-scripts build ./src/ChartsSection/index.js --output-path=build/section",
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/package.json:39]: Why are `start:sections` & `build:sections` excluded from `start` and `build` commands respectively?

File: wp-content/plugins/hurumap/package.json [hunk 10]
```diff
@@ -82,4 +84,4 @@
       "git add"
     ]
   }
-}
+}
```
Comment (kilemensi) [inline wp-content/plugins/hurumap/package.json]: `\n`

**Review by kilemensi — COMMENTED** [review]
:+1: @karimkawambwa 

--------------------------------

1. Flourish charts fail with the unable to unzip file etc. error message,
2. I can create HURUmap chart but I can't include it in a post (same with Flourish),
3. Can create section but can't sort, etc.

File: wp-content/plugins/hurumap/api.php (hunk 4 above)
Comment (karimkawambwa) [reply wp-content/plugins/hurumap/api.php]: Not sure why we do this @KhadijaMahanga ?

File: wp-content/plugins/hurumap/api.php (hunk 2 above)
Comment (karimkawambwa) [reply wp-content/plugins/hurumap/api.php]: Resolved with the `sectioned` query param

**Pull request merged**